src/lib/support/jsontlv/TlvToJson.cpp

```diff
@@ -108,14 +108,12 @@ struct JsonObjectElementContext
         {
             if (TLV::ProfileIdFromTag(tag) == implicitProfileId)
             {
-                // Explicit assume implicit tags are just things we want
-                // 32-bit numbers for
                 str = std::to_string(TLV::TagNumFromTag(tag));
             }
             else
             {
-                // UNEXPECTED, create a full 64-bit number here
-                str = std::to_string(TLV::ProfileIdFromTag(tag)) + "/" + std::to_string(TLV::TagNumFromTag(tag));
+                uint32_t tagNumber = (static_cast<uint32_t>(TLV::VendorIdFromTag(tag)) << 16) | TLV::TagNumFromTag(tag);
```

Comment:
The cast should be on the whole subexpression, not just the the first part.

Reply:
TLV::VendorIdFromTag(tag) returns a uint16_t, not casting before the shift would mean we are shifting within a 16-bit space. The cast of TLV::VendorIdFromTag(tag) to uint32_t before the shift operation ensures that the shift operation occurs on a 32-bit integer.